lib/rototiller/task/params/env_var.rb

```diff
@@ -43,123 +40,79 @@ def initialize(args={}, &block)
-        end
-        if get_message_type == MESSAGE_TYPES[:nodefault_noexist]
-          return red_text("#{message_prepend} is required: #{@user_message}")
+        message = String.new
+
+        if env_status    == STATUS[:nodefault_noexist]
+          message << red_text('ERROR: env_var not set and no default provided: ')
+          message << "'#{@name}': '#{@message}'\n"
+        elsif env_status == STATUS[:nodefault_exist]
+          message << yellow_text('INFO: env_var using system value, no default provided: ')
+          message << "'#{@name}': '#{@value}': '#{@message}'\n"
+        elsif env_status == STATUS[:default_noexist]
+          message << green_text('INFO: env_var with no system value, using default provided: ')
+          message << "'#{@name}': '#{@value}': '#{@message}'\n"
+        elsif env_status == STATUS[:default_exist]
+          message << yellow_text('INFO: env_var overridden from system, not using default: ')
+          message << "'#{@name}': default: '#{@default}' using: '#{@value}': '#{@message}'\n"
         end
+
       end
 
       # The string representation of this EnvVar; the value on the system, or nil
       # @return [String] the EnvVar's value
       def to_str
-        ENV[@name]
+        @value
       end
       alias :to_s :to_str
 
-      # If any of these variables are assigned a new value after this object's creation, reset @value and @user_message_level.
-      def var=(var)
-        @name = var
-        reset
-      end
-
-      # If any of these variables are assigned a new value after this object's creation, reset @value and @user_message_level.
-      def default=(default)
-        @default = default
-        reset
-      end
-
-      # If any of these variables are assigned a new value after this object's creation, reset @value and @user_message_level.
-      def message=(message)
-        @user_message = message
-        reset
-      end
-
-      # If any of these variables are assigned a new value after this object's creation, reset @value and @user_message_level.
-      def required=(required)
-        @required = required
-        reset
-      end
-
-      def set_env=(set=true)
-        @set_env = set
+      # Sets the name of the EnvVar
+      # @raise [ArgumentError] if name contains an illegal character for bash environment variable
+      def name=(name)
+        name.each_char do |char|
+          message = "You have defined an environment variable with an illegal character: #{char}"
+          raise ArgumentError.new(message) unless char =~ /[a-zA-Z]|\d|_/
+        end
+        @name = name
       end
 
       private
 
       # @private
-      #TODO cleanup WTF
       def reset
+        (env_value_provided_by_user? || @default) ? @stop = false : @stop = true
+
         if @name
           @value = ENV[@name] || @default
-          ENV[@name] = @value if @set_env
-          set_message_level
+          if !ENV[@name]
```

Comment:
Should you use the helper method here with an unless? `unless env_value_provided_by_user?`